src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,190 @@
+
+        return result;
+    }
+
+    /**
+     * Gets OTP queries to check non-realtime itinerary existence for the given trip.
+     */
+    public static Map<String, String> getItineraryExistenceQueries(MonitoredTrip trip, boolean checkAllDays) throws URISyntaxException {
+        return getQueriesFromDates(
+            excludeRealtime(trip.parseQueryParams()),
+            getDatesToCheckItineraryExistence(trip, checkAllDays)
+        );
+    }
+
+    /**
+     * @return a copy of the specified query parameter map, with ignoreRealtimeUpdates set to true.
+     */
+    public static Map<String, String> excludeRealtime(Map<String, String> params) {
+        Map<String, String> result = new HashMap<>(params);
+        result.put(IGNORE_REALTIME_UPDATES_PARAM, "true");
+        return result;
+    }
+
+    /**
+     * Replaces the trip itinerary field value with one from the verified itineraries queried from OTP.
+     * The ui_activeItinerary query parameter is used to determine which itinerary to use,
+     * TODO/FIXME: need a trip resemblance check to supplement the ui_activeItinerary param used in this function.
+     */
+    public static void updateTripWithVerifiedItinerary(MonitoredTrip trip, List<Itinerary> verifiedItineraries) throws URISyntaxException {
+        Map<String, String> params = trip.parseQueryParams();
+        Itinerary itinerary = null;
+        String itineraryIndexParam = params.get("ui_activeItinerary");
+        if (itineraryIndexParam != null) {
+            try {
+                int itineraryIndex = Integer.parseInt(itineraryIndexParam);
+                itinerary = verifiedItineraries.get(itineraryIndex);
+            } catch (NumberFormatException e) {
+                // TODO: error message.
+            }
+        } else {
+            // FIXME: implement itinerary resemblance to find one matching trip.itinerary.
+        }
+
+        if (itinerary != null) {
+            trip.itinerary = itinerary;
+            trip.initializeFromItineraryAndQueryParams();
+        }
+    }
+
+    /**
+     * Checks that the specified itinerary is on the same day as the specified date/time.
+     * @param itinerary the itinerary to check.
+     * @param date the request date to check.
+     * @param time the request time to check.
+     * @param checkArrival true to check the itinerary endtime, false to check the startTime.
+     * @return true if the itinerary's startTime or endTime is one the same day as the day of the specified date and time.
+     */
+    public static boolean isSameDay(Itinerary itinerary, String date, String time, ZoneId zoneId, boolean checkArrival) {
+        // TODO: Make SERVICE_DAY_START_HOUR an optional config parameter.
+        final int SERVICE_DAY_START_HOUR = 3;
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern(DEFAULT_DATE_FORMAT_PATTERN);
+
+        ZonedDateTime startDate = ZonedDateTime.ofInstant(itinerary.getStartOrEndTime(checkArrival).toInstant(), zoneId);
```

Comment:
The `Itinerary#getStartOrEndTime` is only ever called here. I'd recommend just extracting the logic and inline it here.